HIP/hip-0000-cross-ledger-protocol-clpr.md

```diff
@@ -0,0 +1,1051 @@
+
+The signature on the hash of the bytes of the connector registration message
+must be verifiable with the public key provided in the `connector_key`.  
+Including the `local_clpr_endpoint_id` is how the clpr endpoint knows that
+the connector owner is intending for the connector to be installed on this
+endpoint.
+
+The `local_clpr_endpoint_type` indicates the content and data format for the
+`local_clpr_endpoint_config`. The configuration content is specific to the
+local CLPR endpoint type and may include parameters for approving the local
+submission of CLPR Application Messages to the connector, charging fees for
+submitted messages, and how to pay for the handling of remotely submitted
+CLPR Application messages.
+
+The `remote_clpr_endpoint_type` indicates the CLPR Endpoint type of the
+remote ledger which entails the data format of the content received from
+the remote endpoint and the type of state proof used to validate the content.
+The `remote_clpr_endpoint_config` indicates the ip addresses to connect
+to when reaching out to the remote endpoint and the initial configuration for
+validating state proofs. This collection of data should be stored in its own
+state and shared through a state proof to the remote ledger to solicit
+updates to the configuration of the remote ledger. The `clpr_endpoint_id`
+is necessary in all connector registrations. The `proof_config` is required
+in the first connector registered to the remote CLPR Endpoint, but may be
+omitted in subsequent connector registrations to the same CLPR Endpoint. If
+the proof configuration for the remote endpoint was bad in the first
+connector registered, additional proof configurations may be provided
+through updates to the connector or additional connector registrations. If the
+list of ip addresses for an endpoint is empty in all connector registrations,
+the local ledger will only receive connections from the remote ledger. If a
+pair of CLPR Endpoints has a connector registered between them, at least one of
+the CLPR Endpoints must be configured with the IP Addresses of the other
+endpoint otherwise neither endpoint will attempt to communicate to the other.
+
+#### Abstract CLPR Channel Protocol
+
+When one CLPR Endpoint (local) reaches out to a remote CLPR Endpoint, there
+is at least 1 registered connector on the local endpoint that has provided
+one or more initial IP addresses to connect to. It is assumed that all IP
+addresses in a connector registration reach the same CLPR Endpoint. Once two
+CLPR Endpoints have established a channel of communication between them, the
+configuration for the remote endpoints is updated through the endpoints
+directly talking to each other through a channel. The process of
+communicating through a channel between two CLPR Endpoints to exchange connector
+state and queued messages is as follows:
+
+1. The initiating endpoint sends a `Proof of Remote CLPR Endpoint Config`
```

Comment:
how does this map to GRPC? seems like it is a series of calls, would be nice if it could be single call with all data and getting all data needed as a response. Or could be streaming. Just feels little wrong to have to orchestrate an exact flow of stateful calls.